ada/ast.py

```diff
@@ -1664,6 +1657,37 @@ def child_decl_initial_env_name(private_part=(T.Bool, False)):
             No(T.Symbol)
         )
 
+    @langkit_property(return_type=T.DesignatedEnv)
+    def child_decl_initial_env(private_part=(T.Bool, False)):
+        """
+        Return the initial env for this basic declaration. This is used
+        to set the parent environment of a child declaration to its actual
+        parent in terms of Ada semantics.
+
+        If ``private_part`` is True, return the env of the private part of its
+        parent.
+        """
+        return Self.child_decl_initial_env_name(private_part).then(
+            lambda name: named_env(name),
+            default_val=direct_env(Self.default_initial_env)
+        )
+
+    @langkit_property(return_type=T.env_assoc)
+    def child_decl_env_assoc():
+        """
+        Return the env association that describes where to register this
+        basic declaration. For child declaration in particular, this orders
```

Comment:
For child declarations* (or “For a child declaration”)

Reply:
Thanks!